File: leetcode/hash-table/49-Group-Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>>mpp;
        for(auto& it:strs){
            string temp = it;
            sort(temp.begin(),temp.end());
            mpp[temp].push_back(it);

        }

        vector<vector<string>>ans(mpp.size());
        int j=0;
        for(auto& it:mpp){

            for(auto& k:it.second){
                ans[j].push_back(k);
            }
            j++;
        }
        return ans;
    }
};

File: leetcode/hash-table/49-Group-Anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,List<String>>mpp = new HashMap<>();

        for(String s:strs){
            // String temp = s;
            // Collections.sort(temp);
            char[] chars = s.toCharArray();
            Arrays.sort(chars);

            String temp = new String(chars);

            // if(!mpp.containsKey(temp)){
            //     mpp.put(temp,new ArrayList<>());
            // }

            // mpp.get(temp).add(s);
            mpp.computeIfAbsent(temp, k -> new ArrayList<>()).add(s);
        }
        List<List<String>> ans = new ArrayList<>(mpp.values());
       
        return ans;
    }
}